packages/headless/src/features/insight-user-actions/insight-user-actions-slice.test.ts

```diff
@@ -0,0 +1,201 @@
+import {insightUserActionsReducer} from './insight-user-actions-slice';
+import {getInsightUserActionsInitialState} from './insight-user-actions-state';
+
+describe('insight user actions slice', () => {
+  const requestId = 'some-request-id';
+  const errorResponse = {
+    message: 'something bad happened',
+    statusCode: 400,
+    type: 'badluck',
+  };
+  const session = {
+    start: '2023-01-04T20:05:13.741Z',
+    end: '2023-01-04T21:00:42.741Z',
+    actions: [
+      {
+        actionType: UserActionType.CLICK,
+        timestamp: '2022-12-04T20:05:13.741Z',
+        eventData: {},
+        cause: 'resultClick',
+        searchHub: 'someSearchHub',
+        document: {
+          title: 'How to do that thing I need to do',
+          clickUri: 'https://www.insightfulwebsite.com/how-to',
+        },
+      },
+    ],
+  };
+
+  it('should have an initial state', () => {
+    expect(insightUserActionsReducer(undefined, {type: 'foo'})).toEqual(
+      getInsightUserActionsInitialState()
+    );
+  });
+
+  describe('#fetcUserActions', () => {
+    it('should set #loading to #true when fetching the user actions', () => {
+      const modifiedState = insightUserActionsReducer(
+        getInsightUserActionsInitialState(),
+        fetchUserActions.pending(requestId)
+      );
+
+      expect(modifiedState.loading).toBe(true);
+    });
+
+    it('should clear the #error when fetching the user actions', () => {
+      const errorState = {
+        ...getInsightUserActionsInitialState(),
+        error: errorResponse,
+      };
+
+      const modifiedState = insightUserActionsReducer(
+        errorState,
+        fetchUserActions.pending(requestId)
+      );
+
+      expect(modifiedState.error).toBeUndefined();
+    });
+
+    describe('when fetching user actions fails', () => {
+      const failedAction = fetchUserActions.rejected(
+        null,
+        requestId,
+        null as unknown as void,
+        errorResponse
+      );
+
+      it('should set #loading to #false', () => {
+        const modifiedState = insightUserActionsReducer(
+          getInsightUserActionsInitialState(),
+          failedAction
+        );
+
+        expect(modifiedState.loading).toBe(false);
+      });
+
+      it('should set #error', () => {
+        const modifiedState = insightUserActionsReducer(
+          getInsightUserActionsInitialState(),
+          failedAction
+        );
+
+        expect(modifiedState.error).toStrictEqual(errorResponse);
+      });
+
+      it('should set #timeline to #undefined', () => {
+        const modifiedState = insightUserActionsReducer(
+          getInsightUserActionsInitialState(),
+          failedAction
+        );
+
+        expect(modifiedState.timeline).toBeUndefined();
+      });
+    });
+
+    describe('when fetching the user actions succeeds', () => {
+      const response: InsightUserActionsResponse = {
+        timeline: {
+          precedingSessions: [session],
+          session: session,
```

Comment:
Is that really expected that the API returns the same session twice?

Reply:
No not at all I just reused the fixture.